PWGMM/Tasks/vertexing-fwd.cxx

```diff
@@ -0,0 +1,198 @@
+  std::vector<double> vecZposCollForAmb; // vector for z vertex of collisions associated to an ambiguous track
+
+  Configurable<float> maxDCAXY{"maxDCAXY", 3.0, "max allowed transverse DCA"}; // To be used when associating ambitrack to collision using best DCA
+
+  HistogramRegistry registry{
+    "registry",
+    {{"TracksDCAXY", "; DCA_{xy} (cm); counts", {HistType::kTH1F, {{100, -1, 10}}}},
+     {"TracksDCAX", "; DCA_{x} (cm); counts", {HistType::kTH1F, {{100, -10, 10}}}},
+     {"TracksDCAY", "; DCA_{y} (cm); counts", {HistType::kTH1F, {{100, -10, 10}}}},
+     {"CollisionsSize", "; Collisions size", {HistType::kTH1F, {{10, -0.5, 9.5}}}},
+     {"NumberOfContributors", "; N_{tr} for vertexing; counts", {HistType::kTH1F, {{100, 0, 100}}}},
+     {"CollisionsMatchIndicesMC", "; Rec. minDCA ambitrack coll.ID; Gen. ambitrack coll.ID", {HistType::kTH2F, {{401, -0.5, 1000.5}, {401, -0.5, 1000.5}}}},
+     {"TracksDCAXYBest", "; DCA_{xy}^{best} (cm); counts", {HistType::kTH1F, {{100, -1, 10}}}},
+     {"TracksDCAXYOther", "; DCA_{xy}^{best, false} (cm); counts", {HistType::kTH1F, {{100, -1, 10}}}},
+     {"TracksDCAXYBestFalse", "; DCA_{xy}^{best, false} (cm); counts", {HistType::kTH1F, {{100, -1, 10}}}},
+     {"EfficiencyZvtx", "; z vertex; Efficiency", {HistType::kTProfile, {{100, -30, 30}}}},
+     {"DeltaZvtx", "; #delta z (cm); counts", {HistType::kTH1F, {{400, -20, 20}}}},
+     {"DeltaZvtxBest", "; #delta z = z_{best} - z_{true} (cm); counts", {HistType::kTH1F, {{400, -20, 20}}}},
+     {"DeltaDCAminNcoll", "; N_{coll}; #delta DCA = DCA_{xy}^{min} - DCA_{xy}^{other} (cm); counts", {HistType::kTH2F, {{10, -0.5, 9.5}, {100, -0.5, 9.5}}}},
+     {"CorrectMatch", "; Matching value; counts", {HistType::kTH1F, {{4, -0.5, 3.5}}}}}};
+
+
+  void process(aod::AmbiguousMFTTracks const& ambitracks, aod::BCs const& bcs, soa::Join<o2::aod::MFTTracks, aod::McMFTTrackLabels> const& tracks, soa::Join<aod::Collisions, aod::McCollisionLabels> const& collisions, aod::McParticles const& mcParticles, aod::McCollisions const& mcCollisions)
+  {
+    for (auto& ambitrack : ambitracks) {
+      vecCollForAmb.clear();
+      vecDCACollForAmb.clear();
+      vecAmbTrack.clear();
+      vecZposCollForAmb.clear();
+
+      double value = 0.0;    // matching value for collision association to an ambiguous track
+      double zVtxMCAmbi = 0; // z vertex associated to the mc collision
+      int mcCollAmbiID = -1; // mc value for the collision containing the ambiguous track
+
+      if (!ambitrack.bc().size() | !ambitrack.globalIndex()) {
+        continue;
+      }
+      if (!tracks.size() | (tracks.size() <= 1)) {
+        continue;
+      }
+      if ((ambitrack.globalIndex() >= tracks.size()) | (ambitrack.globalIndex() < 0)) {
+        continue;
+      }
+
+      // soa::Join<o2::aod::FwdTracks, o2::aod::FwdTracksCov, aod::McFwdTrackLabels>::iterator extAmbiTrack;
```

Comment:
This is not needed, the index should be directly accessible like this: `auto track = ambitrack.mfttrack_as<soa::Join<o2::aod::MFTTracks, aod::McMFTTrackLabels>>();`

One can also introduce an alias `using MFTTracksLabeled = soa::Join<o2::aod::MFTTracks, aod::McMFTTrackLabels>;` and use it to save some typing.

Reply:
Thanks!